src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -1136,10 +1138,12 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger? errorLogger, Cancella
                 {
                     // At this point we have a compilation with nothing yet computed.
                     // We pass it to the generators, which will realize any symbols they require.
-                    (compilation, generatorTimingInfo) = RunGenerators(compilation, Arguments.ParseOptions, generators, analyzerConfigProvider, additionalTextFiles, diagnostics);
+                    var explicitGeneratedOutDir = Arguments.GeneratedFilesOutputDirectory;
+                    var hasExplicitGeneratedOutDir = !string.IsNullOrWhiteSpace(explicitGeneratedOutDir);
+                    var baseDirectory = hasExplicitGeneratedOutDir ? explicitGeneratedOutDir! : Arguments.OutputDirectory;
```

Comment:
>: Arguments.OutputDirectory; [](http://example.com/codeflow?start=94&length=28)

> "When no /generatedfilesout argument is specified, use the OutputDirectory (directory name of /out argument) as the base path for generated files. (Generated files are still only written to disk when /generatedfilesout is explicitly specified.)"

When we're not writing them to disk, is the fact that we're using a different base path for generated syntax trees observable? 
In other words, which test would be affected by the removal of `: Arguments.OutputDirectory`?